modulus/launch/utils/checkpoint.py

```diff
@@ -74,6 +74,7 @@ def _get_checkpoint_filename(
     # can save their checkpoint. In the case without model parallelism,
     # model_parallel_rank should be the same as the process rank itself and
     # only rank 0 saves
+    DistributedManager.initialize()
```

Comment:
This isn't the right fix here I think. This will cause other issues if the user already initialized the DistributedManager. Maybe a guard here would be better:
```python
if not DistributedManager.is_initialized():
    logger.warn("`DistributedManager` not initialized already. Initializing now, but this might lead to unexpected errors")
    DistributedManager.initialize()
```

Reply:
Good call, let me test this

Reply:
Updated